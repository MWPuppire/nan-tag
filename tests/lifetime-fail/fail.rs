use nan_tag::*;

fn main() {
    let tagged;
    {
        let x = 17;
        let y = &x;
        tagged = TaggedNan::new_pointer(y);
    }
    tagged.extract();
}
