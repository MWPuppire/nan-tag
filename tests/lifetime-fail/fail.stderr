error[E0597]: `x` does not live long enough
  --> tests/lifetime-fail/fail.rs:7:17
   |
6  |         let x = 17;
   |             - binding `x` declared here
7  |         let y = &x;
   |                 ^^ borrowed value does not live long enough
8  |         tagged = TaggedNan::new_pointer(y);
9  |     }
   |     - `x` dropped here while still borrowed
10 |     tagged.extract();
   |     ------ borrow later used here
